source/watchdog.py

```diff
@@ -70,6 +87,128 @@ def __getattr__(attrName: str) -> Any:
 _cancelCallEvent = None
 
 
+def _getCurrentCrashFingerprint() -> tuple[str, str]:
```

Comment:
could all this new crash handling code be moved to its own file?

Reply:
@seanbudd  LMK if I'm misusing git by mentioning when replying not sure if I somehow put you back in the review list somehow and say my turn is over. It worked differently at Google. I can split this out if you think it necessary, although see above regarding unnecessary imports. We'd probably need to make watchdog a dir, with a __init__ which might go over better as a separate refactor because it's a bit more moduler.

Reply:
this adds about 100 lines without touching existing code. it's a lot easier to maintain if its more modular. adding imports isn't really added complexity, separating code into separate imports imrpoves maintainability